lib/core/src/msParam.cpp

```diff
@@ -1022,6 +1022,28 @@ parseMspForPosInt( msParam_t* inpParam ) {
     return myInt;
 }
 
+int parseMspForStr_new(msParam_t* inpParam, char** outParam)
+{
+    if (inpParam == NULL || inpParam->inOutStruct == NULL) {
+        rodsLog(LOG_ERROR, "%s: inpParam type %s is not STR_MS_T", __func__, inpParam->type);
+        *outParam = NULL;
+        return 1;
+    }
+
+    if (strcmp(inpParam->type, STR_MS_T) != 0) {
+        rodsLog(LOG_ERROR, "%s: inpParam type %s is not STR_MS_T", __func__, inpParam->type);
+        return 1;
+    }
+
+    if (strcmp((char*) inpParam->inOutStruct, "null") == 0) {
+        *outParam = NULL;
+        return 0;
+    }
+
+    *outParam = (char*) (inpParam->inOutStruct);
+    return 0;
+}
```

Comment:
As a convention, we add a trailing comment containing the name of the function. For example:
```c++
} // parseMspForStr_new
```

That helps with understanding where you are in the code. Especially when functions become very long and cannot be viewed in full within your editor.